### PR TITLE
support akka configs in InitJoinAck

### PR DESCRIPTION
**Pull request by pjfanning**
relates to #1866 

I've been focusing on the InitJoin message but @regiskuckaertz has pointed out that configs are also exchanged in the InitJoinAck message.

I've tried to add test coverage for this change. I think I need to get some of our tests to do Akka-only checks in the JoinCluster checks when certain configs are set up. At the moment the Pekko JoinCluster checks expect joining nodes to have pekko configs and in mixed cluster setup, this is relaxes to allow pekko or akka. It is probably useful in some of our cluster tests to make this stricter and only accept akka configs. In this case, we would be more likely to catch issues with us not adjusting our configs when interacting with Akka nodes.

### CODE REVIEW

**Comment by regiskuckaertz** [comment]
happy to test this one as soon as it's merged ✌️

File: remote/src/main/resources/reference.conf [hunk 1]
```diff
@@ -188,14 +188,21 @@ pekko {
...
     # A ConfigurationException will be thrown at runtime if the value is not
     # set to "pekko" or "akka".
     protocol-name = "pekko"
 
+    # When receiving join requests or related acks from other remote nodes,
+    # should we ensure the config details that are passed in these messages
+    # are prefixed with the protocol name (i.e. "pekko" or "akka")?
+    # The default is "off" as the accept-protocol-names check should be sufficient.
+    # This setting was created to enable edge case tests and users should not change it.
+    enforce-strict-config-prefix-check-on-join = off
```
Comment (He-Pin) [inline remote/src/main/resources/reference.conf:204]: on joining？

**Review by He-Pin — APPROVED** [review]
Lgtm

File: cluster/src/main/scala/org/apache/pekko/cluster/ConfigUtil.scala [hunk 2]
```diff
@@ -19,15 +19,21 @@ package org.apache.pekko.cluster
...
   private val PekkoPrefix = "org.apache.pekko"
   private val AkkaPrefix = "akka"
 
-  @nowarn("msg=deprecated")
   def addAkkaConfig(cfg: Config, akkaVersion: String): Config = {
+    val newConfig = changePekkoToAkkaConfig(cfg)
+    newConfig.withValue("akka.version", ConfigValueFactory.fromAnyRef(akkaVersion))
+  }
+
+  def changePekkoToAkkaConfig(cfg: Config): Config = {
```
Comment (He-Pin) [inline cluster/src/main/scala/org/apache/pekko/cluster/ConfigUtil.scala]: adapte？

File: remote/src/main/resources/reference.conf (hunk 1 above)
Comment (pjfanning) [reply remote/src/main/resources/reference.conf:204]: join is a noun in English (as well as a verb) - as a native English speak, joining would sound less natural than join.

on joining might appear as the start of a sentence or clause but not really at the end

File: cluster/src/main/scala/org/apache/pekko/cluster/ConfigUtil.scala (hunk 2 above)
Comment (pjfanning) [reply cluster/src/main/scala/org/apache/pekko/cluster/ConfigUtil.scala]: added a9cf909f34d5c2dd7a12729e21ce564ac59d810e

**Comment by pjfanning** [comment]
> happy to test this one as soon as it's merged ✌️

@regiskuckaertz 1.2.0-M1+75-d7f0d510-SNAPSHOT contains this change

**Comment by regiskuckaertz** [comment]
rock'n'roll

**Comment by regiskuckaertz** [comment]
Boo-ya 💥 

Node 1 (pekko, advertising itself as akka):
```
org.apache.pekko.cluster.Cluster Cluster(pekko://PTS) - Cluster Node [akka.ssl.tcp://PTS@***:1600] - Starting up, Pekko version [1.2.0-M1+75-d7f0d510-SNAPSHOT] ...
org.apache.pekko.cluster.Cluster Cluster(pekko://PTS) - Cluster Node [akka.ssl.tcp://PTS@***:1600] - Registered cluster JMX MBean [pekko:type=Cluster]
org.apache.pekko.cluster.Cluster Cluster(pekko://PTS) - Cluster Node [akka.ssl.tcp://PTS@***:1600] - Started up successfully
org.apache.pekko.cluster.Cluster Cluster(pekko://PTS) - Cluster Node [akka.ssl.tcp://PTS@***:1600] - No downing-provider-class configured, manual cluster downing required, see https://pekko.apache.org/docs/pekko/current/typed/cluster.html#downing
c.p.c.c.listener.ClusterListener akka.ssl.tcp://PTS@***:1600/user/$b/clusterListener - Starting Cluster Error detection
org.apache.pekko.cluster.Cluster Cluster(pekko://PTS) - Cluster Node [akka.ssl.tcp://PTS@***:1600] - Received InitJoinAck message from [Actor[akka.ssl.tcp://PTS@***:1603/system/cluster/core/daemon#-925069082]] to [akka.ssl.tcp://PTS@***:1600]
org.apache.pekko.cluster.Cluster Cluster(pekko://PTS) - Cluster Node [akka.ssl.tcp://PTS@***:1600] - Welcome from [akka.ssl.tcp://PTS@***:1603]
c.p.c.c.listener.ClusterListener akka.ssl.tcp://PTS@***:1600/user/$b/clusterListener - Member is Up: Member(akka.ssl.tcp://PTS@***:1601, Up), roles: Set(trm-writer, dc-default) - memberCount: 1
c.p.c.c.listener.ClusterListener akka.ssl.tcp://PTS@***:1600/user/$b/clusterListener - Member is Up: Member(akka.ssl.tcp://PTS@***:1603, Up), roles: Set(tlm-writer, dc-default) - memberCount: 2
c.p.c.c.listener.ClusterListener akka.ssl.tcp://PTS@***:1600/user/$b/clusterListener - Member is Up: Member(akka.ssl.tcp://PTS@***:1600, Up), roles: Set(rdstore-processor, dc-default) - memberCount: 3
o.a.p.c.s.ClusterSingletonManager akka.ssl.tcp://PTS@***:1600/system/sharding/PTS_rdstore-processorCoordinator - Singleton manager starting singleton actor [pekko://PTS/system/sharding/PTS_rdstore-processorCoordinator/singleton]
o.a.p.c.s.ClusterSingletonManager akka.ssl.tcp://PTS@***:1600/user/clientActorMonitor - Singleton manager starting singleton actor [pekko://PTS/user/clientActorMonitor/sharedActor]
o.a.p.c.s.ClusterSingletonManager akka.ssl.tcp://PTS@***:1600/system/sharding/PTS_rdstore-processorCoordinator - ClusterSingletonManager state change [Start -> Oldest]
o.a.p.c.s.ClusterSingletonManager akka.ssl.tcp://PTS@***:1600/user/clientActorMonitor - ClusterSingletonManager state change [Start -> Oldest]
org.apache.pekko.cluster.Cluster Cluster(pekko://PTS) - Cluster Node [akka.ssl.tcp://PTS@***:1600] - is the new leader among reachable nodes (more leaders may exist)
```

Node 2 and Node 3 (akka):
```
akka.cluster.Cluster Cluster(akka://PTS) - Cluster Node [akka.ssl.tcp://PTS@***:1603] - Received InitJoin message from [Actor[akka.ssl.tcp://PTS@***:1600/system/cluster/core/daemon/firstSeedNodeProcess-1#1414432955]] to [akka.ssl.tcp://PTS@***:1603]
akka.cluster.Cluster Cluster(akka://PTS) - Cluster Node [akka.ssl.tcp://PTS@***:1603] - Sending InitJoinAck message from node [akka.ssl.tcp://PTS@***:1603] to [Actor[akka.ssl.tcp://PTS@***:1600/system/cluster/core/daemon/firstSeedNodeProcess-1#1414432955]] (version [2.6.21])
akka.cluster.Cluster Cluster(akka://PTS) - Cluster Node [akka.ssl.tcp://PTS@***:1603] - Node [akka.ssl.tcp://PTS@***:1600] is JOINING, roles [rdstore-processor, dc-default], version [0.0.0]
c.p.c.c.listener.ClusterListener akka.ssl.tcp://PTS@***:1603/user/$b/clusterListener - Member is Up: Member(akka.ssl.tcp://PTS@***:1600, Up), roles: Set(rdstore-processor, dc-default) - memberCount: 3
```

**Comment by regiskuckaertz** [comment]
@pjfanning thanks for all your work on this 🙇 no rush at all but so I can plan internally, do you have a rough ETA for release 1.2?

**Comment by regiskuckaertz** [comment]
ah, I just saw the milestone, maybe it's more reasonable of me to use the next 1.1.x then. However if there's an issue I can help with, I'm happy to.

**Comment by pjfanning** [comment]
I have a backfit PR ready for the 1.1 branch. I can seek to get that merged and do a 1.1.4 release after that.
